internal/infrastructure/oceanwallet/notifications_manager.go

```diff
@@ -0,0 +1,67 @@
+package oceanwallet
+
+import (
+	"context"
+
+	"github.com/tdex-network/tdex-daemon/internal/core/ports"
+	oceanv1alpha "github.com/vulpemventures/ocean/api-spec/protobuf/gen/go/ocean/v1alpha"
+	"google.golang.org/grpc"
+)
+
+type notificationsManagerGrpc struct {
+	client oceanv1alpha.NotificationServiceClient
+}
+
+var _ ports.NotificationManager = (*notificationsManagerGrpc)(nil)
+
+func newNotificationsManagerGrpc(conn *grpc.ClientConn) ports.NotificationManager {
+	return &notificationsManagerGrpc{
+		client: oceanv1alpha.NewNotificationServiceClient(conn),
+	}
+}
+
+func (nm *notificationsManagerGrpc) TxChannel() (chan ports.TxNotification, error) {
+	req := &oceanv1alpha.TransactionNotificationsRequest{}
+	stream, err := nm.client.TransactionNotifications(context.Background(), req)
+	if err != nil {
+		return nil, err
+	}
+
+	txChan := make(chan ports.TxNotification)
+
+	go func() {
+		for {
+			resp, err := stream.Recv()
+			if err != nil {
+				return
+			}
```

Comment:
here the channel should be closed in case the error is EOF, while we should `continue` otherwise, maybe by debug-logging to print the error to stdout.

Reply:
Same thing applies to `UtxoChannel()`